### PR TITLE
Refactor IG Graph API calls

### PR DESCRIPTION
**Pull request by pestevez**
This PR contains the following changes to the IG sample app:

- Refactors the multiple hardcoded URLs into a helper function `buildGraphAPIURL`.
- Adds a configuration option for the Graph API version `GRAPH_API_VERSION ` which will default to the default version configured at the app level.
- Adds a configuration option for the Graph API origin `DEFAULT_GRAPH_API_ORIGIN ` which is meant for Meta developers to test functionality that is gated.
- Updates the `.env.template` file to include the new settings

### CODE REVIEW

File: insta_reels_publishing_api_sample/index.js [hunk 1]
```diff
@@ -22,9 +26,24 @@ const {
...
     REDIRECT_URI,
     APP_ID,
-    API_SECRET
+    API_SECRET,
+    GRAPH_API_ORIGIN,
+    GRAPH_API_VERSION_PATH,
 } = process.env;
 
+const GRAPH_API_BASE_URL = (GRAPH_API_ORIGIN ?? DEFAULT_GRAPH_API_ORIGIN) +
+    (GRAPH_API_VERSION_PATH ?? DEFAULT_GRAPH_API_VERSION_PATH);
```
Comment (anujs3) [inline insta_reels_publishing_api_sample/index.js:35]: Shouldn't there be a `/` between `GRAPH_API_ORIGIN` and `GRAPH_API_VERSION_PATH`?

File: insta_reels_publishing_api_sample/.env.template [hunk 2]
```diff
@@ -4,3 +4,11 @@ API_SECRET=YOUR_APP_SECRET
 REDIRECT_URI=https://localhost:8000/callback
 SESSION_SECRET=RANDOM_SESSION_SECRET_STRING
 PORT=8000
+
+# Optional config setting to change the Graph API version.
+# Leave blank or commented to default to the app's default version
+# GRAPH_API_VERSION_PATH=v18
```
Comment (anujs3) [inline insta_reels_publishing_api_sample/.env.template:10]: Can we change this to `v18.0`?

image

File: insta_reels_publishing_api_sample/index.js (hunk 1 above)
Comment (anujs3) [reply insta_reels_publishing_api_sample/index.js:35]: For example, it should be `https://graph.facebook.com/v18.0/`, not `https://graph.facebook.comv18`

File: insta_reels_publishing_api_sample/.env.template [hunk 3]
```diff
@@ -4,3 +4,11 @@ API_SECRET=YOUR_APP_SECRET
...
+
+# Optional config setting to chage the Graph API host
+# Leave blank or commented to default to the default Graph API host name
+# DEFAULT_GRAPH_API_ORIGIN=https://graph.facebook.com
```
Comment (anujs3) [inline insta_reels_publishing_api_sample/.env.template:14]: Shouldn't this be simply `GRAPH_API_ORIGIN`?

File: insta_reels_publishing_api_sample/index.js [hunk 4]
```diff
@@ -63,7 +82,12 @@ app.get("/login", function (req, res) {
 // Callback route for handling FB OAuth user token And reroute to '/pages'
 app.get("/callback", async function (req, res) {
     const code = req.query.code;
-    const uri = `https://graph.facebook.com/oauth/access_token?client_id=${APP_ID}&redirect_uri=${REDIRECT_URI}&client_secret=${API_SECRET}&code=${code}`;
+    const uri = buildGraphAPIURL('oauth/access_token', {
+        client_id: APP_ID,
+        redirect_uri: REDIRECT_URI,
+        client_secret: API_SECRET,
+        code,
```
Comment (anujs3) [inline insta_reels_publishing_api_sample/index.js:89]: Shouldn't this be `code: code`?

File: insta_reels_publishing_api_sample/index.js [hunk 5]
```diff
@@ -193,8 +220,14 @@ app.post("/uploadReels", async function (req, res) {
     if(typeof locationId === 'undefined') {
         locationId = "";
     }
-    const uploadParamsString = `caption=${caption}&cover_url=${coverUrl}&thumb_offset=${thumbOffset}&location_id=${locationId}&access_token=${req.session.userToken}`;
-    const uploadVideoUri = `https://graph.facebook.com/v14.0/${accountId}/media?media_type=REELS&video_url=${videoUrl}&${uploadParamsString}`;
+    const uploadVideoUri = buildGraphAPIURL(`${accountId}/media`, {
+        media_type: 'REELS',
+        video_url: videoUrl,
+        caption,
```
Comment (anujs3) [inline insta_reels_publishing_api_sample/index.js:226]: We don't have to write this as `caption: caption,`? Is it implied?

**Review by anujs3 — CHANGES_REQUESTED** [review]
left a few comments to consider

File: insta_reels_publishing_api_sample/.env.template (hunk 3 above)
Comment (pestevez) [reply insta_reels_publishing_api_sample/.env.template:14]: Nice catch!

File: insta_reels_publishing_api_sample/index.js (hunk 4 above)
Comment (pestevez) [reply insta_reels_publishing_api_sample/index.js:89]: They are equivalent, I am using the [shorthand notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#property_definitions).

File: insta_reels_publishing_api_sample/index.js (hunk 5 above)
Comment (pestevez) [reply insta_reels_publishing_api_sample/index.js:226]: Yeah, it is implied. If the property name has the same name as the variable you can use the [shorthand notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#property_definitions).

File: insta_reels_publishing_api_sample/index.js (hunk 1 above)
Comment (pestevez) [reply insta_reels_publishing_api_sample/index.js]: Tweaked to ensure the `/` are included.

File: insta_reels_publishing_api_sample/.env.template (hunk 2 above)
Comment (pestevez) [reply insta_reels_publishing_api_sample/.env.template]: Changed. I had missed it, thanks!

File: insta_reels_publishing_api_sample/index.js (hunk 5 above)
Comment (anujs3) [reply insta_reels_publishing_api_sample/index.js:226]: Oh that's convenient! Thanks for explaining!

**Review by anujs3 — APPROVED** [review]

**Pull request merged**